src/Models/uml_method.py

```diff
@@ -49,26 +49,19 @@ def add_param (self, p_name:str) -> None:
        self._params.append(UML_Param(p_name))
 
     def delete_param (self, p_name:str) -> None:
-        ''' Deletes a param if it exists
-            Raises: ValueError if p_name is not a parameter of this method
-        '''
-
-        p = self.__find_param(p_name)
-        if p == None:
-            raise ValueError ("No param named %s exists" % p_name)
-        self._params.remove(p)
+        self._params.remove(self.get_param(p_name))
     
     def change_params (self, *p_names) -> None:
-        ''' Replaces the current parameter list with a new list of params '''
+        ''' Replaces the current parameter list with a new list of params ''' 
         self._params.clear()
         self.append_params(*p_names)
     
     def append_params (self, *p_names) -> None:
```

Comment:
If the purpose of line 62 is to just to remove potential duplicates, you could use a set rather than importing the OrderedDict structure

Reply:
Sets don't maintain order, using a set would make the order of the params change randomly every time you change multiple.

Reply:
Pretty sure using a Set was giving us nondeterminism within the list, hence the need for the Ordered part. Peter can probably clarify more.

Reply:
Gotcha, I was thinking ordered e.g. alphabetical not insertion order. Seems like ordered dictionary works best here since ordered set doesn't exist.